src/web/auth.rs

```diff
@@ -62,11 +55,14 @@ impl HawkPayload {
         expiry: u64,
     ) -> AuthResult<HawkPayload> {
         if &header[0..5] != "Hawk " {
-            return Err(AuthError);
+            return Err(HawkErrorKind::MissingPrefix.into());
         }
 
         let header: HawkHeader = header[5..].parse()?;
-        let id = header.id.as_ref().ok_or(AuthError)?;
+        let id = header.id.as_ref().ok_or_else(|| {
+            let error: ApiError = HawkErrorKind::MissingId.into();
+            error
```

Comment:
Here's an example of where the double-layer of `From` meant I had to explicitly annotate the interim type.